narwhal/consensus/src/consensus.rs

```diff
@@ -337,3 +337,102 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::ConsensusState;
+    use fastcrypto::{bls12381::min_sig::BLS12381KeyPair, traits::KeyPair};
+    use rand::thread_rng;
+    use types::Certificate;
+
+    #[test]
+    pub fn test_gc() {
+        let mut state = ConsensusState::new();
+        assert_eq!(state.dag.len(), 0);
+        let kp1 = BLS12381KeyPair::generate(&mut thread_rng());
+        let pub_key1 = kp1.public();
+        let kp2 = BLS12381KeyPair::generate(&mut thread_rng());
+        let pub_key2 = kp2.public();
+        let mut cert1_kp1 = Certificate::new_test_empty(pub_key1.clone());
+        cert1_kp1.header.round = 1;
+        let mut cert1_kp2 = Certificate::new_test_empty(pub_key2.clone());
+        cert1_kp2.header.round = 1;
+        let mut cert2_kp1 = Certificate::new_test_empty(pub_key1.clone());
+        cert2_kp1.header.round = 2;
+        let mut cert2_kp2 = Certificate::new_test_empty(pub_key2.clone());
+        cert2_kp2.header.round = 2;
+        let mut cert3_kp1 = Certificate::new_test_empty(pub_key1.clone());
+        cert3_kp1.header.round = 3;
+        let mut cert3_kp2 = Certificate::new_test_empty(pub_key2.clone());
+        cert3_kp2.header.round = 3;
+        let mut cert4 = Certificate::new_test_empty(pub_key1.clone());
+        cert4.header.round = 4;
+        let mut cert5 = Certificate::new_test_empty(pub_key1.clone());
+        cert5.header.round = 5;
+        state.try_insert(&cert1_kp1);
```

Comment:
seems like a change of pace here; I think it would be nice to separate this test with comments describing what is currently being done

Reply:
not sure what you mean?

Reply:
seems like at the very beginning, we're creating authority keypairs; then, we simulate them creating certificates, and finally, we insert them; it would be nice if these steps were accompanied by comments outlining the general process